src/renderer/reducers/settings.js

```diff
@@ -275,7 +275,10 @@ const handlers: Object = {
     { payload }: { payload: { lastSeenDevice: DeviceModelInfo, latestFirmware: any } },
   ) => ({
     ...state,
-    lastSeenDevice: payload.lastSeenDevice,
+    lastSeenDevice: {
+      ...(state.lastSeenDevice || {}),
+      ...payload.lastSeenDevice,
```

Comment:
could you double check that we don't call this dispatch somewhere with potential undefined :grimacing: 
or maybe fail safe it the same way as the line above `|| {}`

Reply:
you're right, won't do any harm to add it